File: 0101-symmetric-tree/0101-symmetric-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    bool isTreeSymmetric(TreeNode* left, TreeNode* right) {
        if(left == nullptr && right == nullptr)
            return true;
        if(left == nullptr && right != nullptr)
            return false;
        if( left != nullptr && right == nullptr)
            return false;
        if(left -> val != right -> val)
            return false;
        return isTreeSymmetric(left -> left, right -> right)
            && isTreeSymmetric(left -> right, right -> left);
        
    }
    bool isSymmetric(TreeNode* root) {
        return isTreeSymmetric(root -> left, root -> right);
    }
};

File: 0101-symmetric-tree/0101-symmetric-tree.ts
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     val: number
 *     left: TreeNode | null
 *     right: TreeNode | null
 *     constructor(val?: number, left?: TreeNode | null, right?: TreeNode | null) {
 *         this.val = (val===undefined ? 0 : val)
 *         this.left = (left===undefined ? null : left)
 *         this.right = (right===undefined ? null : right)
 *     }
 * }
 */

function isSymmetric(root: TreeNode | null): boolean {
    if(!root) return true;
    let q1 = [root.left], q2 = [root.right];
    while (q1.length > 0 && q2.length > 0) {
        let node1 = q1.shift()
        let node2 = q2.shift()

        if(!node1 && !node2) continue

        if(!node1 || !node2 || node1.val != node2.val) return false

        q1.push(node1.left)
        q1.push(node1.right)
        q2.push(node2.right)
        q2.push(node2.left)
    }

    return true
};